### ISSUE TITLE
BlitzCache Errors on specifying Database in Azure Managed Instance

### ISSUE THREAD

**Issue by agentKnipe** [BrentOzarULTD/SQL-Server-First-Responder-Kit#2353]
**Version of the script**
SELECT @Version = '7.94', @VersionDate = '20200324';

**What is the current behavior?**
the script errors stating that Azure SQL DB does not allow you to change databases.

**If the current behavior is a bug, please provide the steps to reproduce.**
Using Azure Managed Instance
Execute the script with the @Database parameter set to a different database than that which is hosting the script

**What is the expected behavior?**
The script will execute cross database since managed instance allows changing databases.

**Which versions of SQL Server and which OS are affected by this issue? Did this work in previous versions of our procedures?**
Azure Managed Instance, unsure whether this has worked previously on managed instance.

I can say that if I comment out the lines that raise the exception, the script executes as expected.

![image](https://user-images.githubusercontent.com/7571082/80225310-e5258580-8618-11ea-8fd8-fb8641823164.png)

**Comment by erikdarlingdata** [comment]
The code that checks that is `IF SERVERPROPERTY('EngineEdition') IN (5, 6, 8) AND DB_NAME() <> @DatabaseName`. I don't have anything in Azure to help on this one, unfortunately, but is there anything else in `SERVERPROPERTY` that identifies Managed Instances we could add to the check?

**Comment by agentKnipe** [comment]
There appears to be something in sp_Blitz that can identify the managed instance.
![image](https://user-images.githubusercontent.com/7571082/80500560-5a0bfe80-893c-11ea-9e70-230757de66e8.png)

Looks like its on line 473 in sp_blitz
![image](https://user-images.githubusercontent.com/7571082/80500768-95a6c880-893c-11ea-8c38-b7f4a9c50ce9.png)

**Comment by BrentOzar** [comment]
Fixed - and by fixed I mean I just stopped checking for Azure SQL Managed Instances, and it just runs anyway. No guarantees on compatibility with Managed Instances, of course - like we say on the home page of the repo, we don't actually support Azure SQL DB of any kind, just because Microsoft changes the DMV contents without warning and without documentation. This will at least get you started though.
